sr_robot_commander/test/test_robot_commander.py

```diff
@@ -0,0 +1,652 @@
+from sr_robot_commander.sr_robot_commander import SrRobotCommander, MoveGroupCommander, PlanningSceneInterface
+from geometry_msgs.msg import Pose, PoseStamped
+from control_msgs.msg import FollowJointTrajectoryActionGoal
+from moveit_msgs.msg import RobotState, RobotTrajectory
+from moveit_msgs.srv import GetPositionFK, SaveRobotStateToWarehouse
+from trajectory_msgs.msg import JointTrajectoryPoint, JointTrajectory
+from std_msgs.msg import Header
+from sensor_msgs.msg import JointState
+from moveit_commander.exception import MoveItCommanderException
+from moveit_commander import conversions
+from actionlib_msgs.msg import GoalStatusArray
+import tf2_ros
+import time
+
+# Some of the test cases are consisting of 1 line. In case of these methods the test verifies if
+# the API of moveit_commander changed - i.e. change of methods name, number of arguments, return type
+
+PKG = "sr_robot_commander"
+CONST_RA_HOME_ANGLES = {'ra_shoulder_pan_joint': 0.00, 'ra_elbow_joint': 2.00,
+                        'ra_shoulder_lift_joint': -1.57, 'ra_wrist_1_joint': -0.73,
+                        'ra_wrist_2_joint': 1.57, 'ra_wrist_3_joint': 0.00}
+
+CONST_EXAMPLE_TARGET = {'ra_shoulder_pan_joint': 0.2, 'ra_elbow_joint': 1.80,
+                        'ra_shoulder_lift_joint': -1.37, 'ra_wrist_1_joint': -0.52,
+                        'ra_wrist_2_joint': 1.57, 'ra_wrist_3_joint': 0.00}
+
+
+class TestSrRobotCommander(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        rospy.wait_for_message("/move_group/status", GoalStatusArray)
+        cls.robot_commander = SrRobotCommander("right_arm")
+        cls.robot_commander.set_planner_id("RRTstarkConfigDefault")
+        cls.robot_commander.set_planning_time(3)
+        cls.eef = cls.robot_commander.get_end_effector_link()
+
+    def reset_to_home(self):
+        self.robot_commander._reset_plan()
+        plan = self.robot_commander.plan_to_joint_value_target(CONST_RA_HOME_ANGLES, angle_degrees=False,
+                                                               custom_start_state=None)
+        self.robot_commander.execute_plan(plan)
+
+    def compare_poses(self, pose1, pose2, tolerance=0.01):
+        p1_list = [pose1.position.x, pose1.position.y, pose1.position.z,
+                   pose1.orientation.x, pose1.orientation.y, pose1.orientation.z, pose1.orientation.w]
+        p2_list = [pose2.position.x, pose2.position.y, pose2.position.z,
+                   pose2.orientation.x, pose2.orientation.y, pose2.orientation.z, pose2.orientation.w]
+        p1_list = [round(i, 2) for i in p1_list]
+        p2_list = [round(i, 2) for i in p2_list]
+
+        for i in range(0, len(p1_list)):
+            if abs(p1_list[i]) - abs(p2_list[i]) > tolerance:
+                return False
+        return True
+
+    def compare_joint_states(self, joint_state_1, joint_state_2, tolerance=0.01):
+        common_joint_names = set(joint_state_1.keys()).intersection(set(joint_state_2.keys()))
```

Comment:
so I get why this is done, but my quesiton is - do we want to consider two joint states that have different set of joints equal if just the shared joints are equal. As in, if the keys are different, doesnt it make the joint_state different in it self? Just asking, not saying.

Reply:
Well, I guess basically the best scenario is to consider both - keys and values to be equal and rename to are_equal_joint_states(), if I'd rename my method to compare_joint_states_by_common_keys() would it make more sense for you?